espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/web/WebAtomDeserializer.kt

```diff
@@ -19,22 +21,27 @@ class WebAtomDeserializer : JsonDeserializer<WebAtom> {
             val jsonObj = json.asJsonObject
 
             // Parse the name of the Atom
-            AssertHelpers.assertIsPrimitive(jsonObj, "name");
+            AssertHelpers.assertIsPrimitive(jsonObj, "name")
             val webAtomName = jsonObj.get("name").asString
 
             // Parse the Locator shorthand. If it was provided.
             jsonObj.get("locator")?.let {
                 // Validate the locator
                 if (!it.isJsonObject) {
-                    throw JsonParseException("'locator' must be an object with properties 'using' and 'value'");
+                    throw JsonParseException("'locator' must be an object with properties 'using' and 'value'")
                 }
 
                 // Validate that the locator contains "using" and "value"
-                val locator = it.asJsonObject;
+                val locator = it.asJsonObject
                 if (!locator.has("using") || !locator.has("value")) {
-                    throw JsonParseException("'locator' must have properties 'using' and 'value'");
+                    throw JsonParseException("'locator' must have properties 'using' and 'value'")
                 }
 
+                val locatorsMapping = Locator.values().fold(mutableMapOf<String, Locator>(), { acc, item ->
+                    acc[item.name.replace("_", "")] = item
```

Comment:
Is this necessary?
Possible breaking change for `CSS_SELECTOR` and `PARTIALLINKTEXT` etc

Reply:
I was not quite sure which strategy was applied before. Did it expect snake_case or camelCase locator type names? I don't mind implementing any of these or even both of them

Reply:
Ah, okay. Locator names were snake_case with upper case. https://developer.android.com/reference/androidx/test/espresso/web/webdriver/Locator